lib/beaker/dsl/helpers.rb

```diff
@@ -564,6 +564,8 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
 
           yield self if block_given?
 
+        rescue Beaker::DSL::Assertions, Minitest::Assertion => early_assertion
```

Comment:
I believe that you only need to rescue Beaker::DSL::Assertions, as they include Minitest::Assertions.